spec/services/replication_transfer_updater_spec.rb

```diff
@@ -0,0 +1,266 @@
+          update.update
+          expect(record.store_requested).to be false
+        end
+      end
+    end
+  end
+
+
+  describe ReplicationTransferUpdater::StoreRequestedUpdate do
+    let(:update) { described_class.new(record, updated) }
+    let(:record) { Fabricate(:replication_transfer, store_requested: false) }
+    let(:changes) { {store_requested: true} }
+    describe "::matching_update?" do
+      it "matches old.store_requested == false, new.store_requested == true" do
+        expect(described_class.matching_update?(record, updated)).to be true
+      end
+    end
+    describe "#update" do
+      shared_examples "store_requested_update" do
+        it "updates the record" do
+          update.update
+          expect(record.store_requested).to be true
+        end
+        it "returns true on success" do
+          allow(record).to receive(:update).and_return true
+          expect(update.update).to be true
+        end
+        it "returns false on failure" do
+          allow(record).to receive(:update).and_return false
+          expect(update.update).to be false
+        end
+        it "does not cancel the record" do
+          update.update
+          expect(record.cancelled).to be false
+        end
+        it "does not set stored" do
+          update.update
+          expect(record.stored).to be false
+        end
+      end
+      context "we are not replicating" do
+        let(:record) { Fabricate(:replication_transfer, store_requested: false, to_node: Fabricate(:node)) }
+        it_behaves_like "store_requested_update"
+        it "does not request preservation" do
+          bag_man_request = double(:bag_man_request)
+          allow(record).to receive(:bag_man_request).and_return(bag_man_request)
+          expect(bag_man_request).to_not receive(:okay_to_preserve!)
+          update.update
+        end
+      end
+      context "we are replicating" do
+        let(:record) { Fabricate(:replication_transfer, store_requested: false, to_node: Fabricate(:local_node)) }
+        it_behaves_like "store_requested_update"
+        it "requests preservation" do
+          bag_man_request = double(:bag_man_request)
+          allow(record).to receive(:bag_man_request).and_return(bag_man_request)
+          expect(bag_man_request).to receive(:okay_to_preserve!)
+          update.update
+        end
+      end
+    end
+  end
+
+
+  describe ReplicationTransferUpdater::StoredUpdate do
+    let(:update) { described_class.new(record, updated) }
+    let(:changes) { {stored: true} }
+    describe "::matching_update?" do
+      let(:record) { Fabricate(:replication_transfer, stored: false) }
+      it "matches old.stored == false, new.stored == true" do
+        expect(described_class.matching_update?(record, updated)).to be true
+      end
+    end
+    describe "#update" do
+      context "we did not request this transfer" do
+        let(:record) { Fabricate(:replication_transfer, stored: false) }
```

Comment:
Could this be explicit about setting the `from_node` to something and setting the `to_node` to something else, which might be local_node or another node altogether.  This prompts me to think there are actually 3 contexts:
- `local_node` == `from_node` (!= `to_node`) - so all the admin node operations apply
- `local_node` == `to_node` (!= `from_node`) - so all the replicating node operations apply
- `local_node` is not either `to_node or `from_node` - this should do only a generic update